lib/logstash/plugin_mixins/validator_support/required_host_optional_port_validation_adapter.rb

```diff
@@ -13,7 +13,7 @@ module ValidatorSupport
       is_valid_ipv4 = ->(candidate) { Resolv::IPv4::Regex.match?(candidate) && Resolv::IPv4.create(candidate) rescue false}
       is_valid_ipv6 = ->(candidate) { Resolv::IPv6::Regex.match?(candidate) && Resolv::IPv6.create(candidate) rescue false}
 
-      is_valid_hostname = ->(candidate) { (%r{\A([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}\Z}i).match?(candidate) }
+      is_valid_hostname = ->(candidate) { (%r{\A([a-z0-9]+(-[a-z0-9]+)*\.)?[a-z]{2,}\Z}i).match?(candidate) }
```

Comment:
This pattern feels a bit arbitrary, both too lax (`this..that` passes but should fail) and not enough (`asd-f` fails, but should pass).

I wrote and [validated](https://regexr.com/7n2ce) an RFC1123-compliant expression that enforces the format of the labels, the label max lengths, and the whole hostname max length, all without wildcards that could lead to backtracking issues:

```suggestion
      # RFC1123 validated at https://regexr.com/7n2ce
      rfc1123_hostname_pattern = %r{\A(?=.{1,255}\z)(?:(?=[^.]{1,63}(?:[.]|\z))[a-z0-9]+(?:[a-z0-9\-]+[a-z0-9])?(?:[.]|\z))+\z}i
      is_valid_hostname = ->(candidate) { rfc1123_hostname_pattern.match?(candidate) }
```